### PR TITLE
Fix posttest fetch slow issue

### PR DESCRIPTION
**Pull request by auspham**
### Description of PR

Summary: Fix post test fetch file issue for large dump file
Fixes # (issue) 29889214

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
There is a huge performance impact when using ansible fetch with `become=True`. This is also documented in the ansible module documentation (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/fetch_module.html) where they suggested not to use `become=True`. 

However in our case we need to use `become=True` for fetch because we're trying to fetch from `/var/dump`

The same issue was also reported in https://github.com/ansible/ansible/issues/31194. 

#### How did you do it?
The solution is to create a fetch module without dependent on slurp. Doing this will improve 600 times the speed

#### How did you verify/test it?
Testing with 1.4GB file, in nightly and manual test we're expecting 4 hours to finish downloading this file using normal `fetch`
Using `fetch_no_slurp` this will take 1.5 minutes

Verified on different T0, T2 platform.

### CODE REVIEW

File: ansible/plugins/action/fetch_no_slurp.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+from ansible.utils.display import Display
+from ansible.utils.hashing import checksum, md5, secure_hash
+from ansible.utils.path import makedirs_safe, is_subpath
+
+display = Display()
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
```
Comment (yejianquan) [inline ansible/plugins/action/fetch_no_slurp.py:19]: Why do we need a tmp here? Is it an official signature designed by Ansible?

Comment (auspham) [reply ansible/plugins/action/fetch_no_slurp.py:19]: Yes this is official signature from their method interface. I believe this is for backward compatibility usage: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/__init__.py#L91

FYI: most of the code here is identical to https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/fetch.py with some modification to remove their use of slurp and force use `fetch_file`

**Review by yejianquan — APPROVED** [review]
LGTM

**Comment by auspham** [comment]
possibly related: https://github.com/sonic-net/sonic-mgmt/issues/15152

**Pull request merged**